### PR TITLE
Samuel Yeo

### PR DESCRIPTION
**Pull request by SK-Sam**
Didn't have a better idea to check for objects so I used attributes of the objects to check if it's in the right position in the array it's in.

### CODE REVIEW

**Comment by SK-Sam** [comment]
I failed, I forgot to add the hoarded portion of Iteration 1, oh god

File: lib/shopping_cart.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  def search_for_nums(string_input) # "30item"
+    valid_numbers = "1234567890"
+    only_nums_string = ""
+    input_to_verify = string_input.split('')
+    input_to_verify.each do |char|
+      only_nums_string += char if valid_numbers.include?(char)
+      return only_nums_string.to_i if !valid_numbers.include?(char)
+    end
+    only_nums_string.to_i
+  end
```
Comment (megstang) [inline lib/shopping_cart.rb:19]: Wowza!! Nice job on this.. definitely not necessary, but great work.

File: lib/shopping_cart.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    {
+      name: @name,
+      capacity: capacity
+    }
+  end
+
+  def total_number_of_products
+    @products.reduce(0) do |sum, product|
+      sum + product.quantity
+    end
```
Comment (megstang) [inline lib/shopping_cart.rb:41]: nice use of reduce

File: lib/shopping_cart.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+  def percentage_occupied
+    total_number_of_products / capacity.to_f * 100
+  end
+
+  def sorted_products_by_quantity
+    @products.sort_by do |product|
+      product.quantity
+    end
+  end
```
Comment (megstang) [inline lib/shopping_cart.rb:62]: reverse this return value

File: lib/shopping_cart.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  def get_list_of_unique_product_categories
+    list_of_product_categories = @products.map do |product|
+      product.category
+    end.uniq.sort
+  end
+
+  def product_by_breakdown
+    product_list = Hash.new
+    get_list_of_unique_product_categories.each do |category|
+      product_list[category] = products_by_category(category)
```
Comment (megstang) [inline lib/shopping_cart.rb:73]: I like how you're using this products by category here

**Review by megstang — COMMENTED** [review]
@SK-Sam Good job on this. You missed a few methods in I1, but overall good job! I left a few comments, please let me know if you have any question.

**Comment by SK-Sam** [comment]
Thanks as always @megstang for the feedback and keen eye to where I need to work on. I'll take everything to face value and push for more accuracy along with your fixes!

As always, much appreciated :)

**Pull request closed without merging**